### PR TITLE
ATV sorting

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: src/mempool.cpp [hunk 1]
```diff
@@ -41,13 +38,19 @@ void MemPool::uploadVbkContext(const VTB& vtb) {
   }
 }
 
+void MemPool::uploadVbkContext(const ATV& atv) {
+  block_index_[atv.containingBlock.getHash()
+                   .trimLE<vbk_prev_block_hash_size>()] = atv.containingBlock;
+
+  for (const auto& b : atv.context) {
+    block_index_[b.getHash().trimLE<vbk_prev_block_hash_size>()] = b;
```
Comment (Warchant) [inline src/mempool.cpp:46]: instead of using trim, define a function `short_hash_t makeShortHash(hash_t)`

Comment (Mr-Leshiy) [reply src/mempool.cpp]: fixed.

File: src/mempool.cpp [hunk 2]
```diff
@@ -130,69 +173,71 @@ std::vector<AltPopTx> MemPool::getPop(const AltBlock& current_block,
...
         popTxs.push_back(popTx);
       }
-      stored_atvs_.erase(atv_it);
-      atv_it = stored_atvs_.begin();
-      continue;
     }
 
-    ++atv_it;
+    atvs.pop_front();
+    std::make_heap(atvs.begin(), atvs.end(), atv_comparator);
```
Comment (blackyblack) [inline src/mempool.cpp:217]: Why make_heap again?

File: include/veriblock/entities/vtb.hpp [hunk 3]
```diff
@@ -89,6 +89,10 @@ struct VTB {
   friend bool operator==(const VTB& a, const VTB& b) {
     return a.getId() == b.getId();
   }
+
+  friend bool operator!=(const VTB& a, const VTB& b) {
+    return !(a.getId() == b.getId());
```
[the remark this answers is not in the document]
Comment (Mr-Leshiy) [reply include/veriblock/entities/vtb.hpp:94]: fixed

File: src/mempool.cpp (hunk 2 above)
Comment (Mr-Leshiy) [reply src/mempool.cpp]: fixed

**Pull request merged**